packages/engine/src/engine.ts

```diff
@@ -149,41 +144,97 @@ export function createChart(args: CreateChart): Chart {
  */
 export function nearestScorableNote(
   input: Input,
-  chart: Chart
+  chart: GameChart
```

Comment:
There is a bug in this logic - play to the end of a song, like Abyss Breaker (5) and you cannot hit the last few notes. We need to fix this.